blogs/2022/09/15/dev-container-features.md

```diff
@@ -0,0 +1,141 @@
+
+Now that we’ve seen how to create a Feature, how can I distribute it to others?
+
+## Distribution
+
+Features are distributed as tarballs. The tarball contains the entire contents of the Feature sub-directory, including the `devcontainer-feature.json`, `install.sh`, and any other files in the directory.
+
+The [Open Container Initiative](https://opencontainers.org/), aka OCI, defines industry standards for containers and container resources. We treat Features as OCI Artifacts and use the concept of an [OCI Registry](https://containers.dev/implementors/features-distribution/#oci-registry) to distribute Features.
+
+The [Features template repository](https://github.com/devcontainers/feature-template) mentioned above includes a [GitHub Actions workflow](https://github.com/marketplace/actions/dev-container-publish) to automate the publishing process. It packages each feature into a tarball and publishes the assets as an OCI artifact to GHCR. Trigger the `release.yaml` workflow from the template repository by selecting it on the left of the repository’s Actions tab on GitHub. This will publish each feature to GHCR under the `<owner>/<repo>` namespace. A Feature is only republished when the version property in its `devcontainer-feature.json` is updated.
+
+> **Note:** One manual step with GHCR is to mark the OCI [package as "public"](https://github.com/devcontainers/feature-template#marking-feature-public). This only has to be done once per Feature. Private features do not require this step and may be accessed as long as you’ve logged into the Docker CLI using the credentials for your registry.
+
+### Sharing your Features with the community
+
+If you'd like your contributions to appear in the [VS Code Remote-Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) or [GitHub Codespaces](https://github.com/features/codespaces) UI for dev container creation, you may do the following:
+
+* Go to [devcontainers.github.io](https://github.com/devcontainers/devcontainers.github.io) (the GitHub repo backing [containers.dev](https://containers.dev))
+* Open a PR to modify the [`collection-index.yml`](https://github.com/devcontainers/devcontainers.github.io/blob/gh-pages/_data/collection-index.yml) file
+
+Once merged, your change will appear at [containers.dev/collections](https://containers.dev/collections).
+
+![Feature collections page of containers.dev](./features-collections.png)
+
+## What if my Feature should only install after another one?
+As a Feature author, you may find that your Feature should install before or after other Features. In your `devcontainer-feature.json`, you may use the `installsAfter` property to list Features that should execute before it.
```

Comment:
Perhaps we could mention here to _first_ try without the property (we want the devcontainer cli to be able to intelligently pick an order).  This is a good property to use in cases where ordering explicitly matters, and the CLI won't be able to figure it out.

Reply:
Good call out. I added a note:

> **Note:** We recommend to first creating your Feature without `installsAfter` - the dev container CLI will attempt to intelligently pick an order to execute Features. But in cases where ordering explicitly matters and the CLI won't be able to figure it out, `installsAfter` is a good choice.

Reply:
Any cases where CLI ordering isn't deterministic, and things would sometimes work and sometimes not?

Reply:
Described [here](https://containers.dev/implementors/features/#installation-order):

> By default, Features are installed on top of a base image in an order determined as optimal by the implementing tool.
> 
> If any of the following properties are provided in the Feature’s devcontainer-feature.json, or the user’s devcontainer.json, the order indicated by these propert(ies) are respected (with decreasing precedence).
> 
> 1. The overrideFeatureInstallOrder property in user’s devcontainer.json. Allows users to control the order of execution of their Features.
> 2. The installsAfter property defined as part of a Feature’s devcontainer-feature.json.

So tools implementing the spec, like Remote-Containers or Codespaces, may add logic to determine an optimal order.

Reply:
Removed the initial note mentioned above and instead added the content I listed here, as I think it's clearer.